src/reducers/AddSiteReducer.js

```diff
@@ -10,44 +12,63 @@ import {
     SITE_ACCESSIBILITY_OPTION_CHANGED,
     SITE_FACILITIES_OPTION_CHANGED,
     ADD_SITE_FIELDS_RESET,
-    CURRENT_LOCATION_UPDATED
+    CURRENT_LOCATION_UPDATED,
+    CHECK_IF_SITE_IS_READY,
+    ADD_SITE_SUCCESS,
+    ADD_SITE_FAILURE
 } from '../actions/types';
 
-import {campsite} from '../locale.en';
+import {campsite, reducerAlerts} from '../locale.en';
 
 const {campsite_form: {accessibility_options, facilities_options}} = campsite;
 
 const INITIAL_STATE = {
+    addSiteModalVisible: false,
     latitudeText: '',
+    readyLatitude: 0,
     longitudeText: '',
+    readyLongitude: 0,
     siteTitleText: '',
     siteDescriptionText: '',
     siteDirectionsText: '',
     siteNearestTownText: '',
     accessibilityOption: accessibility_options.paved_road,
-    facilitiesOption: facilities_options.full_service
+    facilitiesOption: facilities_options.full_service,
+    siteReadyForUpload: false
 };
 
 const removeNonNumbers = (text) => {
     return text.replace(/[^0-9.-]/g, '');
 };
 
+const siteIsReadyForUpload = ({latitudeText, longitudeText, siteTitleText, siteDescriptionText, siteDirectionsText, siteNearestTownText, accessibilityOption, facilitiesOption}) => {
+    return latitudeText && longitudeText && siteTitleText && siteDescriptionText && siteDirectionsText && siteNearestTownText && accessibilityOption && facilitiesOption;
+};
```

Comment:
Helper function that just checks to see if all properties of a site are truthy. This is used to toggle whether the "Submit site" button appears on the `AddSiteScreen` modal.